docs/workers.md

```diff
@@ -445,6 +447,10 @@ Specify its name in the shared configuration as follows:
 update_user_directory_from_worker: worker_name
 ```
 
+It can handle REST endpoint matching the following regular expression:
+
+    ^/_matrix/client/(r0|v3|unstable)/user_directory/search$
+
```

Comment:
I think it's fine to omit this part, since the user directory worker _is_ a generic worker.
```suggestion
```

Reply:
Well, many other places has it on both places, on the long list of generic workers and in the specific section of that something... This would be consistent with that. and it also would denote that especially this worker especially for this endpoint is desired.

Reply:
I mean, I'm generally fine with either way, but some notion on specific section is warranted in my mind, it isn't as clear when reading this documentation.. Other than that I'm generally not biased either way on this specific writing.

Reply:
The ones which appear in both places have to be routed to specific workers. I thought this endpoint could be handled by any worker? Down below, the section for `synapse.app.user_dir` says:

```
Above endpoint is not *required* to be routed to this worker. By default,
`update_user_directory` is set to `true`, which means the main process
will handle updates. All workers configured with `client` can handle the above
endpoint as long as either this worker or the main process are configured to
handle it, and are online.

If `update_user_directory` is set to `false`, and this worker is not running,
the above endpoint may give outdated results.
```

Reply:
Sure, endpoint _can_ be processed by any worker, but if one is going to do worker especially for handling user directory stuff it makes only sense to remember also point this endpoint to it. This is my reasoning in mentions :)

Reply:
I see what you're getting at now! Maybe we can word it like so:

```diff
  This work cannot be load-balanced; please ensure the main process is restarted
  after setting this option in the shared configuration!
  
+ User directory updates allow REST endpoints matching the following regular
+ expressions to work:
+ 
+     ^/_matrix/client/(r0|v3|unstable)/user_directory/search$
+ 
+ The above endpoints can be routed to any worker, though you may choose to route
+ it to the chosen user directory worker.
+ 
  This style of configuration supersedes the legacy `synapse.app.user_dir`
  worker application type.
```